### PR TITLE
v0.15 release notes

### PR DESCRIPTION
**Pull request by travisgroth**
## Summary

Update changelog and release notes for v0.15.

Also:

- add a brief writeup of our release process so it doesn't live in my head
- tweak changelog script so it doesn't cache in the repo directory

## Related issues

n/a

## Checklist

- [ ] reference any related issues
- [x] updated docs
- [ ] updated unit tests
- [x] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [ ] ready for review

### CODE REVIEW

File: docs/docs/upgrading.md [hunk 1]
```diff
@@ -5,6 +5,15 @@ description: >-
   for Pomerium. Please read it carefully.
 ---
 
+# Since 0.14.0
+
+## New
```
Comment (travisgroth) [inline docs/docs/upgrading.md:10]: @desimone @alexfornuto I stubbed out new features that could use highlighting. 

1. Are there others that make sense?
2. We need to document the new `routes` syntax and PPL in OSS and link from here.  Can be a FUP once that's ready.

Comment (desimone) [reply docs/docs/upgrading.md:10]: I think we'll have to merge #2390 before linking to the new PPL / Routes sections which are currently WIP.

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Overall process
+
+1. Add new branch to netlify build settings (new release branch only)
+2. Request new subdomain for netlify TLS certificate (new release branch only)
+3. Generate changelog
+4. Update docs/upgrading.md for major/minor releases
+5. Update vuepress default branch
```
Comment (desimone) [inline RELEASING.md:7]: How does one do that?

Comment (travisgroth) [reply RELEASING.md]: Added a little more detail

File: docs/docs/upgrading.md [hunk 3]
```diff
@@ -5,6 +5,20 @@ description: >-
...
+
+### Updated and expanded policy syntax
+
+Routes and policies may now be configured under a new top level key - `routes`.
+
+- This more closely aligns to how policies and routes are conceptually related.
+- The `routes` block supports a more powerful syntax for definiting policies with conditionals and various criteria
+### Support environmental proxy settings
```
[the remark this answers is not in the document]
Comment (travisgroth) [reply docs/docs/upgrading.md:22]: Added a missing space between sections.  Is there something else I'm not seeing?

**Comment by travisgroth** [comment]
Rebased onto master.

File: docs/docs/upgrading.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (desimone) [reply docs/docs/upgrading.md:22]: The newline before the paragraph block, sorry about that I should have been more specific!

File: RELEASING.md (hunk 2 above)
Comment (desimone) [reply RELEASING.md]: Thanks!

**Review by desimone — APPROVED** [review]

File: docs/docs/upgrading.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (travisgroth) [reply docs/docs/upgrading.md:22]: 👍

**Pull request merged**